subprojects/core/src/test/groovy/org/gradle/caching/local/internal/DirectoryBuildCacheEntryExpirationTest.groovy

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.caching.local.internal
+
+import org.gradle.api.internal.cache.CacheConfigurationsInternal
+import org.gradle.api.internal.cache.CacheResourceConfigurationInternal
+import org.gradle.caching.BuildCacheServiceFactory
+import org.gradle.caching.local.DirectoryBuildCache
+import spock.lang.Specification
+
+import java.util.concurrent.TimeUnit
+
+import static org.gradle.internal.time.TimestampSuppliers.daysAgo
+import static org.gradle.internal.time.TimestampSuppliers.inThePast
+
+class DirectoryBuildCacheEntryExpirationTest extends Specification {
```

Comment:
I think we should have an integration test here that checks the build operation, not just a unit test.

Reply:
Agreed.  I reckon we could make it a part of [this test](https://github.com/gradle/gradle/blob/3c455091da1c78144e43f37ecc843136165b2aa7/platforms/core-execution/build-cache/src/integTest/groovy/org/gradle/caching/internal/BuildCacheBuildOperationsIntegrationTest.groovy#L37).  I think `FinalizeBuildCacheConfigurationBuildOperationType` is the build operation type you'd be looking for.